pipeline/transformer/transformer_tei/src/test/scala/weco/pipeline/transformer/tei/TeiXmlTest.scala

```diff
@@ -0,0 +1,18 @@
+package weco.pipeline.transformer.tei
+
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import weco.pipeline.transformer.tei.fixtures.TeiGenerators
+
+class TeiXmlTest extends AnyFunSpec with Matchers with TeiGenerators {
+  val id = "manuscript_15651"
+
+  it("fails parsing a tei XML if the supplied id is different from the id in the XML"){
+    val suppliedId = "another_id"
+    val result = TeiXml(suppliedId, teiXml(id = id).toString())
+    result shouldBe a [Left[_,_]]
+    result.left.get.getMessage should include (suppliedId)
+    result.left.get.getMessage should include (id)
```

Comment:
Can we be a bit more explicit about the message here?

Reply:
I usually prefer not to be too specific about the error message in tests because effectively it's not testing a functionality but testing logging and logs should be easy to change without having to also change the corresponding test IMO. The main aim was to test that whatever the message is, it mentions the two ids because that's probably what we are going to need if we ever have to debug this